src/dash-table/utils/actions.js

```diff
@@ -53,23 +53,31 @@ export const clearSelection = {
     selected_cells: []
 };
 
-export function editColumnName(column, columns, headerRowIndex) {
+export function changeColumnHeader(column, columns, headerRowIndex, mergeDuplicateHeaders, newColumnName) {
+    let cloneColumn = R.mergeRight({}, column);
```

Comment:
There's no need to clone the column if it's not modified in the `if` below. `R.set` below (https://github.com/plotly/dash-table/pull/498/files#diff-4102a065e79334d16000051ec35189f7R75) will treat the entire data structure as immutable in any case.